kernel/kernel-api/src/main/java/io/delta/kernel/internal/DeltaHistoryManager.java

```diff
@@ -162,6 +167,39 @@ public static long getEarliestRecreatableCommit(Engine engine, Path logPath)
         }
     }
 
+    /** Get the persisted commit info (if available) for the given delta file. */
+    public static Optional<CommitInfo> getCommitInfoOpt(
+            Engine engine,
+            Path logPath,
+            long version) throws IOException {
+        final FileStatus file = FileStatus.of(FileNames.deltaFile(logPath, version), 0, 0);
+        final StructType COMMITINFO_READ_SCHEMA = new StructType()
+                .add("commitInfo", CommitInfo.FULL_SCHEMA);
+        final CloseableIterator<ColumnarBatch> columnarBatchIter = engine.getJsonHandler()
+                .readJsonFiles(
+                        singletonCloseableIterator(file),
+                        COMMITINFO_READ_SCHEMA,
+                        Optional.empty());
+        if (!columnarBatchIter.hasNext()) {
+            return Optional.empty();
```

Comment:
We should add logging around places where `CommitInfo` could not be found. Missing `CommitInfo` should be very rare.